build-system/tasks/issues-tasks.js

```diff
@@ -0,0 +1,275 @@
+          if (label.name.startsWith('Type') || label.name.startsWith('Related')) {
+            issueType = label.name;
+          }
+          // Check if the issues has Priority
+          if (label.name.startsWith('P0') || label.name.startsWith('P1') || 
+            label.name.startsWith('P2') || label.name.startsWith('P3')) { 
+            hasPriority = true; 
+          }
+        }
+      });
+      promise = promise.then(function() {
+        util.log('Update ' + issue.number);
+        var updates = [];
+        // Milestone task: move issue from closed milestone
+        if (milestone) {
+          if (milestoneTitle.startsWith('Sprint') && milestoneState == 'closed') {
+            // 4 is the number for Milestone 'Backlog Bugs' 
+            issueNewMilestone = 4; 
+            updates.push(applyMilestone(issue, issueNewMilestone));
+          }
+        }
+        //if issueType is not null, add correct milestones
+        if (issueType != null) {
+          if (milestoneTitle == 'Pending Triage' || milestone == null) {
+            if (issueType == 'Type: Feature Request') {
+              // 23 is the number for Milestone 'New FRs' 
+              issueNewMilestone = 23;
+              updates.push(applyMilestone(issue, issueNewMilestone)); 
+            } else {
+              if (issueType == 'Related to: Documentation' || 
+                issueType == 'Type: Design Review' || issueType == 'Type: Weekly Status') {
+                // 12 is the number for Milestone 'Docs Updates' 
+                issueNewMilestone = 12;
+                updates.push(applyMilestone(issue, issueNewMilestone));
+              } else {
+                if (issueType == 'Type: Bug' || issueType == 'Related to: Flaky Tests') {
+                  // 4 is the number for Milestone 'Backlog Bugs' 
+                  issueNewMilestone = 4;
+                  updates.push(applyMilestone(issue, issueNewMilestone));
+                } else {
+                  if (milestone == null) {
+                    // 20 is the number for Milestone 'Pending Triage' 
+                    updates.push(applyMilestone(issue, issueNewMilestone));
+                  }
+                }
+              }
+            }
+          }
+        } else {
+          if (milestone == null) {
+            // 20 is the number for Milestone 'Pending Triage' 
+            updates.push(applyMilestone(issue, issueNewMilestone));     
+          } else {
+            if (milestoneTitle == 'Prioritized FRs' || milestoneTitle == 'New FRs') {
+              updates.push(applyLabel(issue, 'Type: Feature Request'));
+            } else {
+              if (milestoneTitle == 'Backlog Bugs' || milestoneTitle.startsWith('Sprint')) {
+                updates.push(applyLabel(issue, 'Type: Bug'));
+              }
+            } 
+          }                              
+        }
+        // Apply default priority if no priority
+        if (hasPriority == false && milestoneTitle != 'New FRs' && 
```

Comment:
I'd suggest using `===` and `!==` instead of `==` and `!=` wherever possible, since they avoid type conversions and might help catch bugs that would otherwise slip by unnoticed.

Reply:
ok, changed them for most of them, but for some I need to keep it because the type might be different and I care only about the values (eg. either if it's undefined or null)